vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionManager.java

```diff
@@ -118,11 +125,45 @@ public void updateExhibitionLanguages(Exhibition exhibition, List<String> codes,
                 exhibitionLanguage.setDefault(exhibitionLanguage.getCode().equalsIgnoreCase(defaultLanguage));
             });  
 
-        // Removes exhibition langauge if unselected.
-        exhibition.getLanguages().removeAll(exhibition.getLanguages().stream()
-                .filter(language -> !codes.contains(language.getCode())).collect(Collectors.toList()));
+        // Finds exhibition language if unselected (to be deleted).
+        List<IExhibitionLanguage> exhibitionLanguageToBeRemoved = exhibition.getLanguages().stream()
+                .filter(language -> !codes.contains(language.getCode())).collect(Collectors.toList());
 
+        for (IExhibitionLanguage language  : exhibitionLanguageToBeRemoved ) {
+            if(checkIfLocalizedTextsExists(language))  {
+                throw new ExhibitionLanguageDeletionException() ;
+            }
+
+        }       
+        exhibition.getLanguages().removeAll(exhibitionLanguageToBeRemoved);
     }
+    
+    /**
+     * Return true if given exhibition language has non empty localized texts linked to it
+     * 
+     */
+    @Override
+    public boolean checkIfLocalizedTextsExists(IExhibitionLanguage language)  {               
+        List<LocalizedText> localizedTexts = localizedTextRepo.findByExhibitionLanguage(language);
+        List<LocalizedText> emptyLocalizedTexts = localizedTexts.stream()
+                .filter(localizedText -> !StringUtils.hasText(localizedText.getText())).collect(Collectors.toList());
+        deleteEmptyLocalizedTexts(emptyLocalizedTexts);
+
+        //This returns true if non empty localized texts exist
+        return localizedTexts.size() > emptyLocalizedTexts.size();
+
+    }
+    
+    /**
+     * Removes localized texts from parent entities and delete them.
+     * 
+     */
+    @Override
+    public void deleteEmptyLocalizedTexts(List<LocalizedText> emptyLocalizedTexts) {
```

Comment:
this does not need it's only method. Also, the method name is wrong since it deletes any list given (doesn't matter if they are empty or not).